core/src/main/java/com/sequenceiq/cloudbreak/service/consumption/ConsumptionService.java

```diff
@@ -0,0 +1,156 @@
+import com.sequenceiq.consumption.api.v1.consumption.model.common.ResourceType;
+import com.sequenceiq.consumption.api.v1.consumption.model.request.CloudResourceConsumptionRequest;
+
+@Service
+public class ConsumptionService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConsumptionService.class);
+
+    @Inject
+    private EntitlementService entitlementService;
+
+    @Inject
+    private ConsumptionClientService consumptionClientService;
+
+    @Inject
+    private CmTemplateProcessorFactory cmTemplateProcessorFactory;
+
+    @Value("${cb.consumption.enabled:false}")
+    private boolean consumptionEnabled;
+
+    public void scheduleAttachedVolumeConsumptionCollectionForStackIfNeeded(StackDto stackDto) {
+        executeIfDeploymentFlagAndEntitlementAreSatisfied(stackDto, "attached volume consumption collection scheduling",
+                this::scheduleAttachedVolumeConsumptionCollectionForStack);
+    }
+
+    private void executeIfDeploymentFlagAndEntitlementAreSatisfied(StackDto stackDto, String operation, BiConsumer<StackDto, String> action) {
+        String accountId = stackDto.getAccountId();
+        if (consumptionEnabled && entitlementService.isCdpSaasEnabled(accountId)) {
+            action.accept(stackDto, operation);
+        } else {
+            LOGGER.info("Skipping {} for stack because {}", operation,
+                    consumptionEnabled ? String.format("CDP_SAAS entitlement is missing for account '%s'", accountId) : "it is disabled for the deployment");
+        }
+    }
+
+    private void scheduleAttachedVolumeConsumptionCollectionForStack(StackDto stackDto, String operation) {
+        BiConsumer<StackDto, ConsumptionType> action = (stackDtoAction, consumptionTypeAction) ->
+                scheduleConsumptionCollectionForCloudResourceAndType(stackDtoAction, stackDtoAction.getResourceCrn(), consumptionTypeAction);
+        executeIfCloudPlatformAndStackEligibilityAreSatisfied(stackDto, operation, action);
+    }
+
+    private void executeIfCloudPlatformAndStackEligibilityAreSatisfied(StackDto stackDto, String operation, BiConsumer<StackDto, ConsumptionType> action) {
+        String cloudPlatform = stackDto.getCloudPlatform();
+        ConsumptionType consumptionType = getAttachedVolumeConsumptionType(cloudPlatform);
+        if (consumptionType != null) {
+            if (isStackEligibleForScheduling(stackDto, operation)) {
+                action.accept(stackDto, consumptionType);
+            } else {
+                LOGGER.info("Skipping {} for stack because it is not eligible", operation);
+            }
+        } else {
+            LOGGER.info("Skipping {} for stack because it is not supported for the cloud platform '{}'", operation, cloudPlatform);
+        }
+    }
+
+    private ConsumptionType getAttachedVolumeConsumptionType(String cloudPlatform) {
+        return "AWS".equals(cloudPlatform) ? ConsumptionType.EBS : null;
```

Comment:
I think we should run this only for aws so the check should be on upper level

Reply:
As discussed privately, this was a misunderstanding. The only "higher" level this could be placed would be the entry point of `ConsumptionService`, the calling handler, or perhaps the initiating action. Neither of them seems justified; the `null` value returned here for non-AWS cloud platforms will result in a log message and the abortion of `ConsumptionService`. In particular, no web service call to cloud-consumption will take place in those cases.